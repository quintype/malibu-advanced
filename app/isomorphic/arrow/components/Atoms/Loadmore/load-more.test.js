import * as React from "react";
import renderer from "react-test-renderer";
import { LoadmoreButton } from "./index";
import { Provider } from "react-redux";
import { generateStore, generateCollection } from "../../Fixture";

const collection = generateCollection();

describe("Load more button", () => {
  it("Should render button and load items subsequently with SubsequentLoadCount as template property.", () => {
    const wrapper = renderer
      .create(
        <Provider store={generateStore}>
          <LoadmoreButton template="SubsequentLoadCount" collection={collection} />
        </Provider>
      )
      .toJSON();
    expect(wrapper).toMatchSnapshot();
  });
  it("Should render button and navigate to the corresponding url with NavigateToPage as template property.", () => {
    const wrapper = renderer
      .create(
        <Provider store={generateStore}>
          <LoadmoreButton template="NavigateToPage" collection={collection} />
        </Provider>
      )
      .toJSON();
    expect(wrapper).toMatchSnapshot();
  });
});
