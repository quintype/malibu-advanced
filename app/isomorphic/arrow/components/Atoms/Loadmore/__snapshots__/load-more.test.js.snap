// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Load more button Should render button and load items subsequently with SubsequentLoadCount as template property. 1`] = `
<div
  className="arr--button load-more-m_button__1mmf1 load-more-m_dark__3dfzb load-more-m_default__2_k4V"
  data-test-id="load-more"
>
  Read More
</div>
`;

exports[`Load more button Should render button and navigate to the corresponding url with NavigateToPage as template property. 1`] = `
<a
  aria-label="load-more"
  data-test-id="load-more"
  href="/"
  onClick={[Function]}
>
  <div
    className="arr--button load-more-m_button__1mmf1 load-more-m_dark__3dfzb load-more-m_default__2_k4V"
  >
    Read More
  </div>
</a>
`;
